classlib/src/main/java/org/teavm/classlib/java/lang/TMath.java

```diff
@@ -343,12 +349,14 @@ public static double nextUp(double d) {
         if (TDouble.isNaN(d)) {
             return d;
         }
+        if (d == 0.0d) {
```

Comment:
This change excludes cases for 0.0 and -0.0. After it to determine sign only comparing with zero is needed, but not long bitwise shift with comparing with zero below (so performance should be better IMHO).
I don't think that it will harm C or WASM backend somehow.